src/java.base/share/classes/jdk/internal/classfile/BootstrapMethodEntry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.classfile;
+
+import java.util.List;
+
+import jdk.internal.classfile.constantpool.ConstantPool;
+import jdk.internal.classfile.constantpool.LoadableConstantEntry;
+import jdk.internal.classfile.constantpool.MethodHandleEntry;
+import jdk.internal.classfile.impl.ConcreteBootstrapMethodEntry;
+
+/**
+ * Models an entry in the bootstrap method table.  The bootstrap method table
+ * is stored in the {@code BootstrapMethods} attribute, but is modeled by
+ * the {@link ConstantPool}, since the bootstrap method table is logically
+ * part of the constant pool.
+ */
+public sealed interface BootstrapMethodEntry
```

Comment:
BootstrapMethodEntryis not a constant pool entry, but BootstrapMethodsAttribute entry.
It might be rather moved under attribute package and renamed to BootstrapMethodInfo to follow the same pattern as other attributes/infos.

Reply:
I know it's part of an attribute - but reading the javadoc:
```
/**
 * Models an entry in the bootstrap method table.  The bootstrap method table
 * is stored in the {@code BootstrapMethods} attribute, but is modeled by
 * the {@link ConstantPool}, since the bootstrap method table is logically
 * part of the constant pool.
 */
 ```
 
 And also, seeing the method:
 
 ```
 /**
     * {@return the constant pool associated with this entry}
     */
    ConstantPool constantPool();
```

It seems like the API is doing the (justifiable) trick of making BSMs look like if they are first-class CP entries (even if not specified that way in the JVMLS). I think that's a fine move, but if we go down that path we should be honest, and put this class in constantpool. At least this is my 0.02$.

Reply:
I see your point. The duality of `BootstrapMethodEntry` and potential confusion when moved under `attribute` or under `constantpool` package is the reason why it probably should stay where it is.
However feel free to discuss it at the mailing list.
Thanks.